addons/binding/org.openhab.binding.network/src/main/java/org/openhab/binding/network/internal/dhcp/DHCPListenService.java

```diff
@@ -0,0 +1,69 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.network.internal.dhcp;
+
+import java.net.SocketException;
+import java.util.Map;
+import java.util.TreeMap;
+
+/**
+ * A singleton. IPRequestReceivedCallback objects can register and unregister.
+ * If the first one is registered and there is no singleton instance, an instance will be created and the
+ * receiver thread will be started. If the last IPRequestReceivedCallback is removed, the thread will be stopped
+ * after the receive socket is closed.
+ * IPRequestReceivedCallback will be called for the address that is registered and matches the
+ * DHO_DHCP_REQUESTED_ADDRESS address field.
+ *
+ * @author David Graeff - Initial contribution
+ */
+public class DHCPListenService {
+    static DHCPPacketListenerServer instance;
+    static Map<String, IPRequestReceivedCallback> registeredListeners = new TreeMap<>();
+
+    public static synchronized DHCPPacketListenerServer register(String hostAddress,
+            IPRequestReceivedCallback dhcpListener) throws SocketException {
+        if (instance == null) {
+            instance = new DHCPPacketListenerServer((String ipAddress) -> {
+                IPRequestReceivedCallback listener = registeredListeners.get(ipAddress);
+                if (listener != null) {
+                    instance.logger.info("DHCP request for registered address: {}", ipAddress);
+                    listener.dhcpRequestReceived(ipAddress);
```

Comment:
I need a singleton for DHCP (only one additional socket should listen on the udp port). But I didn't want to make the `DHCPPacketListenerServer` class a singleton because it is harder to test singletons. Therefore this new class `DHCPListenService` creates a single object of the server and allows to register/unregister callbacks. If you have another design idea, please let me now.